gusto/advection.py

```diff
@@ -92,6 +92,9 @@ def _setup(self, state, field, options):
             self.Projector = Projector(self.xdg_out, self.x_projected,
                                        solver_parameters=parameters)
 
+        if options.name == "embedded_dg" and self.limiter is not None:
```

Comment:
does this mean that self.Projector is defined but not used when `options.name=="embedded_dg"` and there is no limiter?

Reply:
Yes you're right -- I could put this under an `else`

Reply:
maybe also find a name that works for both `self.Projector` and `self.x_recoverer` then you can simplify the code in lines 144-147